ui/src/shared/components/AddAnnotationToggle.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {SFC} from 'react'
+import {connect} from 'react-redux'
+
+import {
+  addingAnnotation,
+  dismissAddingAnnotation,
+} from 'src/shared/actions/annotations'
+
+import {ADDING} from 'src/shared/annotations/helpers'
```

Comment:
not even sure, presumably something with nice reference equality properties :)  it was in use before this PR, I didn't change it to avoid changing annotation components that otherwise were working well